src/idl/tests/annotation.c

```diff
@@ -45,6 +45,210 @@ parse_string(uint32_t flags, const char *str, idl_pstate_t **pstatep)
+            CU_ASSERT_EQUAL(def->value.chr, *(const char*)test.default_val_ptr);
+            break;
+          case IDL_STRING:
+            CU_ASSERT_STRING_EQUAL(def->value.str, *(const char**)test.default_val_ptr);
+            break;
+          case IDL_BOOL:
+            CU_ASSERT_EQUAL(def->value.bln, *(const bool*)test.default_val_ptr);
+            break;
+          default:
+            break;
+        }
+      } else {
+        CU_ASSERT_PTR_NULL_FATAL(def);
+      }
+    }
+    idl_delete_pstate(pstate);
+  }
+}
+
+CU_Test(idl_annotation, idl_default)
+{
+  static const int32_t t1 = 123456789;
+  static const double t2 = 987.654321;
+  static const char t3 = 'a';
+  static const bool t4 = true;
+  static const char *t5 = "hello world!";
+  static const idl_default_test_t tests[] = {
+    {"struct s { long l; };",                                IDL_RETCODE_OK,              false, IDL_NULL,    NULL}, //no default whatsoever
+    {"struct s { @default(123456789) long l; };",            IDL_RETCODE_OK,              true,  IDL_LONG,    &t1},  //default long
+    {"struct s { @default(987.654321) double d; };",         IDL_RETCODE_OK,              true,  IDL_DOUBLE,  &t2},  //default double
+    {"struct s { @default('a') char c; };",                  IDL_RETCODE_OK,              true,  IDL_CHAR,    &t3},  //default char
+    {"struct s { @default(true) boolean b; };",              IDL_RETCODE_OK,              true,  IDL_BOOL,    &t4},  //default bool
+    {"struct s { @default(\"hello world!\") string str; };", IDL_RETCODE_OK,              true,  IDL_STRING,  &t5},  //default string
+    {"struct s { @default(123) @optional long l; };",        IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //mixing default and optional
+    {"struct s { @default long l; };",                       IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //misssing parameter
+    {"struct s { @default(123) string str; };",              IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //parameter type mismatch (int vs string)
+    {"struct s { @default(\"false\") boolean b; };",         IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //parameter type mismatch (string vs bool)
+    {"struct s { @default(123) boolean b; };",               IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //parameter type mismatch (int vs bool)
+    {"union u (switch long) {\n"
+     "case 1: @default(123) short s;\n"
+     "case 2: string s; };",                                 IDL_RETCODE_SYNTAX_ERROR,    false, IDL_NULL,    NULL}, //unions do not allow defaults on values
+    /* skipping this test as idl_create_annotation_appl leaks memory if idl_resolve cannot resolve the scoped name (https://github.com/eclipse-cyclonedds/cyclonedds/issues/950)
+      {"@default(e_0) enum e { e_0, e_1, e_2, e_3 };",         IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}  //setting default on enums is done through @default_literal
+    */
```

Comment:
Better enable this and indicate the test is known to fail using the [WILL_FAIL property](https://cmake.org/cmake/help/latest/prop_test/WILL_FAIL.html). Then, when the fix for #950 is in, the test will pass (causing the build to fail). The advantage being that we're actively notified when a change is required.

Reply:
The issue is not here that the test fails, it is that there is a memory leak that is triggered by this test, and that the builds in Azure that check for memory leaks will fail if this test is done.

Reply:
I know, but CTest will mark it as a failure. Since `WILL_FAIL` is a CTest property, it'll get us the desired behavior right? If you prefer you may set the test to disabled. It's just that commenting it out only ensures it's never looked after again. At least, there's tons of commented out tests (or test ideas) for libidl that would've caught a bunch of things but they were never looked at or thought of again... I prefer to have some kind of notification that something is still off, at least that way no one forgets.